src/components/common/UserProfileEdit/UserAdvancedSettings.tsx

```diff
@@ -28,78 +29,74 @@ const MSG = defineMessages({
-    and pay for your own transactions, you can turn them off
-    by switching the toggle at any time. {br}{br} Please note,
-    this setting is stored locally in your browser,
-    if you clear your cache you will need to turn Metatransactions off again.`,
-  },
-  customEndpoints: {
-    id: `${displayName}.customEndpoints`,
-    defaultMessage: `Enable custom endpoints ({isOn, select,
-      true {active}
-      other {inactive}
-    })`,
-  },
-  endpointsDescription: {
-    id: `${displayName}.metaDescription`,
-    defaultMessage: `If you prefer maximum decentralisation, you may use your own custom endpoints for Colony.`,
-  },
-  labelRPC: {
-    id: `${displayName}.labelRPC`,
-    defaultMessage: 'Gnosis Chain RPC',
-  },
-  RPCTooltip: {
-    id: `${displayName}.RPCTooltip`,
-    defaultMessage: `You will be able to toggle the Gnosis Chain RPC on once you have successfully validated that the endpoint works.`,
-  },
-  validate: {
-    id: `${displayName}.validate`,
-    defaultMessage: 'Validate',
-  },
-  snackbarSuccess: {
-    id: `${displayName}.snackbarSuccess`,
-    defaultMessage: 'Profile settings have been updated.',
-  },
   rpcErrorMessage: {
     id: `${displayName}.rpcErrorMessage`,
     defaultMessage: 'Enter a valid URL',
   },
 });
 
-const validationSchema = object()
-  .shape({
-    [SlotKey.Metatransactions]: bool<boolean>(),
-    [SlotKey.DecentralizedMode]: bool<boolean>(),
-    [SlotKey.CustomRPC]: string()
-      .url(() => MSG.rpcErrorMessage)
-      .defined(),
-  })
-  .defined();
+const validationSchema = object({
+  [SlotKey.Metatransactions]: bool<boolean>(),
+  [SlotKey.DecentralizedMode]: bool<boolean>(),
+  [SlotKey.CustomRPC]: string()
+    .defined()
+    .when(`${[SlotKey.DecentralizedMode]}`, {
+      is: true,
+      then: string().url(() => MSG.rpcErrorMessage),
+    }),
+}).defined();
+
+export type FormValues = InferType<typeof validationSchema>;
+
+const AdvancedSettingsHeading = () => (
+  <Heading3
+    appearance={{ theme: 'dark' }}
+    text={MSG.heading}
+    textValues={{
+      learnMoreLink: (
+        <ExternalLink
+          text={{ id: 'text.learnMore' }}
+          href={ADVANCED_SETTINGS}
+          className={stylesAdvance.link}
+          key="learnMore"
+        />
+      ),
+    }}
+  />
+);
+
+// custom isDirty to handle case where user changes CustomRPC but leaves DecentralizedMode off
+const filterCustomRpc = (_, key: string) => key !== SlotKey.CustomRPC;
+const getIsDirty = (
+  values: FormValues,
+  defaultValues: DefaultValues<FormValues>,
+) =>
+  !isEqual(
+    filter(values, filterCustomRpc),
+    filter(defaultValues, filterCustomRpc),
+  );
+
+const setFormValuesToLocalStorage = (
+  values: FormValues,
+  setSettingsKey: UserSettingsHook['setSettingsKey'],
+) => {
+  Object.entries(values).forEach(
+    ([key, value]: [SlotKey, string | boolean]) => {
+      if (
+        // don't save changes to custom rpc if decentralized mode is off
```

Comment:
The thinking here is to prevent someone changing the custom rpc by accident -> for example, with decentalized mode on, they change the rpc url, but then change their mind and turn decentralized mode off and submit the form without reverting the rpc url to its original value.

In this case, presumably they didn't mean to change the rpc url (because why change it only to have decentralized mode off?). So this check would prevent any updates to the custom rpc if decentralized mode is off.

Reply:
I think you're overthinking this :)

So the point of this feature is to be super power user feature. Stuff like "by accident" should not apply here since we're dealing with people who know how to set up, sync and configure their own archive node.

I'm ok with both versions of this interaction, but you shouldn't be too concerned with the UX in this case.

Reply:
Ok, I think simpler is better so have removed